### PR TITLE
Add grid gallery examples

### PR DESCRIPTION
**Pull request by ntouran**

### CODE REVIEW

File: doc/gallery-src/framework/run_grids2_cartesian.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+=====================
+
+This builds a Cartesian grid with squares 1 cm square. with the z coordinates
+provided explicitly. It is also offset in 3D space to X,Y,Z = 10, 5, 5 cm. 
+
+Learn more about :py:mod:`grids <armi.reactor.grids>`.
+"""
+import itertools
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
```
Comment (jakehader) [inline doc/gallery-src/framework/run_grids2_cartesian.py:12]: Unused?

File: doc/gallery-src/framework/run_grids3_rzt.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+=====================
+
+This builds a 3-D grid in Theta-R-Z geometry by specifying the theta, r, and z 
+dimension bounds explicitly. 
+
+Learn more about :py:mod:`grids <armi.reactor.grids>`.
+"""
+import itertools
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
```
Comment (jakehader) [inline doc/gallery-src/framework/run_grids3_rzt.py:12]: Unused?

File: doc/gallery-src/framework/run_grids2_cartesian.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+fig = plt.figure()
+zCoords = [1, 4, 8]
+cartesian_grid = grids.Grid(
+    unitSteps=((1, 0), (0, 1)), bounds=(None, None, zCoords), offset=(10, 5, 5),
+)
+xyz = []
+
+
+for i, j, k in itertools.product(range(10), range(10), range(len(zCoords) - 1)):
```
Comment (jakehader) [inline doc/gallery-src/framework/run_grids2_cartesian.py:28]: Make a comment explaining the ranges here

File: doc/gallery-src/framework/run_grids3_rzt.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+fig = plt.figure()
+theta = np.linspace(0, 2 * np.pi, 10)
+rad = np.linspace(0, 10, 10)
+z = np.linspace(5, 25, 6)
+rz_grid = grids.ThetaRZGrid(bounds=(theta, rad, z))
+
+
+xyz = []
+for i, j, k in itertools.product(range(9), range(9), range(5)):
```
Comment (jakehader) [inline doc/gallery-src/framework/run_grids3_rzt.py:29]: For these ranges, can you just do `for i, j, k in itertools.product(range(len(theta) - 1), range(len(rad) - 1), range(len(z) - 1):`?

File: doc/gallery-src/framework/run_grids2_cartesian.py (hunk 1 above)
Comment (ntouran) [reply doc/gallery-src/framework/run_grids2_cartesian.py:12]: actually it is secretly used.  By importing it it activates the `3d` projection option. If you take it out it fails!

File: doc/gallery-src/framework/run_grids3_rzt.py (hunk 2 above)
Comment (ntouran) [reply doc/gallery-src/framework/run_grids3_rzt.py:12]: same as above. required for 3d

File: doc/gallery-src/framework/run_grids3_rzt.py (hunk 4 above)
Comment (ntouran) [reply doc/gallery-src/framework/run_grids3_rzt.py:29]: yeah

**Pull request merged**